docs/src/userguide/faq.rst

```diff
@@ -0,0 +1,1018 @@
+
+::
+
+    def work_with_text_data(text):
+        if not isinstance(text, unicode):
+            raise ValueError("requires text input, got %s" % type(text))
+        utf8_data = text.encode('UTF-8')
+        c_handle_data( utf8_data, len(utf8_data) )
+
+Note that this also accepts subtypes of the Python unicode type. Typing the "text" parameter as "unicode" will not cover this case.
+
+The above is the right thing to do in Py3. However, some (not all, just some) module APIs may become more user friendly in Python 2.x if you additionally allow well defined byte strings. For example, it may make sense to allow plain ASCII strings in some cases, as they are often used for textual data in Python 2.x programs. This could be done as follows:
+
+::
+
+    from python_version cimport PY_MAJOR_VERSION
+
+    def work_with_text_data(text):
+        if isinstance(text, unicode): # most common case first
+            utf8_data = text.encode('UTF-8')
+        elif (PY_MAJOR_VERSION < 3) and isinstance(text, str):
+            text.decode('ASCII') # trial decoding, or however you want to check for plain ASCII data
+            utf8_data = text
+        else:
+            raise ValueError("requires text input, got %s" % type(text))
+        c_handle_data(utf8_data, len(utf8_data))
+
+----------
+
+How do I use variable args?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Answer**: It can't be done cleanly yet, but the code below works:
+
+::
+
+    cdef extern from "stdarg.h":
+        ctypedef struct va_list:
+            pass
+        ctypedef struct fake_type:
+            pass
+        void va_start(va_list, void* arg)
+        void* va_arg(va_list, fake_type)
+        void va_end(va_list)
+        fake_type int_type "int"
+
+    cdef int foo(int n, ...):
+        print "starting"
+        cdef va_list args
+        va_start(args, <void*>n)
+        while n != 0:
+            print n
+            n = <int>va_arg(args, int_type)
+        va_end(args)
+        print "done"
+
+    def call_foo():
+        foo(1, 2, 3, 0)
+        foo(1, 2, 0)
+
+----------
+
+How do I make a standalone binary from a Python program using cython?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Answer**: You probably want a recipe something like this:
+
+::
+
+    PYVERSION=2.7
+    foobar: foobar.py
+        cython --embed foobar.py -o foobar.c
+        $(CC) -I /usr/include/python$(PYVERSION) foobar.c -lpython$(PYVERSION) -o foobar
+
+The magic is the --embed option, which embeds a copy of the Python interpreter main in the generated C.  You'll want to change 'foobar' to reflect the name of your script, of course, and PYVERSION as appropriate.
+
+----------
+
+How do I have to wrap C code that uses the restrict qualifier?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Answer**: There currently is no way of doing this directly into C code. Cython does not understand the restrict qualifier. However you can wrap your way around it.
```

Comment:
Is there an issue for this? -- I couldn't find one. `restrict` has been in C since C99, so this should IMO be added.